Securitytxt/Model/Config/Backend/Validate.php

```diff
@@ -35,26 +35,40 @@ public function validateBeforeSave()
         $dataGroup = $this->getData()['groups'];
         $contactInformationFields = $dataGroup['contact_information']['fields'];
         $otherInformationFields = $dataGroup['other_information']['fields'];
-        $isExtensionEnabled = (bool)$dataGroup['general']['fields']['enabled']['value'];
-        $contactEmail = $contactInformationFields['email']['value'];
-        $contactPhone = $contactInformationFields['phone']['value'];
-        $contactWebPage = $contactInformationFields['contact_page']['value'];
-
-        if ($isExtensionEnabled) {
-            if ($contactEmail === '' && $contactPhone === '' && $contactWebPage === '') {
-                throw new ValidatorException(__('At least one contact information is required.'));
-            }
-        } else {
-            return parent::validateBeforeSave();
+        $isEnabledField = $dataGroup['general']['fields']['enabled'];
+
+        if ($this->isEnabledDataValue($isEnabledField)
+            && $this->isEmptyContactInformationFields($contactInformationFields)) {
+            throw new ValidatorException(__('At least one contact information is required.'));
         }
 
-        $this->validateContactEmail($contactEmail);
-        $this->validateContactWebpageUrl($contactWebPage);
-        $this->validateUrlField("Contact Page URL", $contactWebPage);
-        $this->validateUrlField("Encryption URL", $otherInformationFields['encryption']['value']);
-        $this->validateUrlField("Acknowledgements URL", $otherInformationFields['acknowledgements']['value']);
-        $this->validateUrlField("Hiring URL", $otherInformationFields['hiring']['value']);
-        $this->validateUrlField("Policy URL", $otherInformationFields['policy']['value']);
+        /**
+         * Validate Email
+         */
+        $this->validateContactEmail($contactInformationFields['email']);
```

Comment:
Will lead to PHP notice undefined key if "email" is not present. The same for the other cases.
Thanks

Reply:
Updated